evap/results/templates/results_course_detail.html

```diff
@@ -21,9 +21,19 @@
     {% endif %}
 
     {% if staff or contributor %}
-        {% if public_view %}
+        {% if course.is_private %}
+            <div class="alert alert-info hidden-print">
+                {% trans "This course is private. Only contributors and participants can see the results." %}
+            </div>
+        {% endif %}
+        {% if public_view and course.is_private %}
+            <div class="alert alert-info hidden-print">{% trans "This is the participant's view. Participants who are not contributors of this course nor their delegates can only see the results below." %}</div>
+            <a href="{% url "results:course_detail" course.semester.id course.id %}?public_view=false" class="pull-right btn btn-sm btn-primary hidden-print">{% trans "Show my view" %}</a>
+        {% elif public_view %}
             <div class="alert alert-info hidden-print">{% trans "This is the public view. Users who are neither contributors of this course nor their delegates can only see the results below." %}</div>
             <a href="{% url "results:course_detail" course.semester.id course.id %}?public_view=false" class="pull-right btn btn-sm btn-primary hidden-print">{% trans "Show my view" %}</a>
```

Comment:
yes, but i don't see a version that doesn't make everything even worse right now. do you have any suggestion?